src/bitcoin-tx.cpp

```diff
@@ -412,6 +445,9 @@ static void MutateTxSign(CMutableTransaction& tx, const string& flagStr)
 
     bool fHashSingle = ((nHashType & ~SIGHASH_ANYONECANPAY) == SIGHASH_SINGLE);
 
+    // Grab the consensus branch ID for the given height
```

Comment:
should this have a +1 ? On that note perhaps we should later refactor the CurrentEpochBranchId to some global function without parameters so we don't need to think about it each time

Reply:
> should this have a +1 ?

Technically no, but practically maybe? The CLI modifications I made now ask the user to provide a height, but it isn't clear what that height might need to be. I'll think about whether I can improve the help text. I did consider passing in a branch ID instead, but wasn't sure that was any more usable.

> On that note perhaps we should later refactor the CurrentEpochBranchId to some global function without parameters so we don't need to think about it each time

- That would not help here, as `zcash-tx` is a standalone contextless binary.
- That would not help in general, as we (and upstream) are trying to move away from depending on globals deep inside functions, instead passing in the necessary context from the caller.